### PR TITLE
Setup rails project and add controllers

### PR DESCRIPTION
**Pull request by DyaryRaoof**
- Created a new app. Gave it a meaningful name.
- Initialize project with Git.
- Project has Postgres database set up.
- App handles the URLs used in the wireframes:
- For each URL I create:
  - A route.
  - An action in the correct controller.
  - A view file.
  - View files include only placeholders written in plain HTML

### CODE REVIEW

File: config/database.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# SQLite. Versions 3.8.0 and up are supported.
+#   gem install sqlite3
+#
+#   Ensure the SQLite 3 gem is defined in your Gemfile
+#   gem "sqlite3"
+#
+default: &default
+  adapter: sqlite3
```
Comment (Lameck1) [inline config/database.yml]: - [ ] According to the requirements, the project should be set up with the PostgreSQL database. The database should be using postgresql in this case. Please refactor the database setup accordingly. The project requirement for this is quoted below: 

 > - Make sure that your project has Postgres database set up.

File: Gemfile [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+ruby '3.0.0'
+
+# Bundle edge Rails instead: gem "rails", github: "rails/rails", branch: "main"
+gem 'rails', '~> 7.0.1'
+
+# The original asset pipeline for Rails [https://github.com/rails/sprockets-rails]
+gem 'sprockets-rails'
+
+# Use sqlite3 as the database for Active Record
+gem 'sqlite3', '~> 1.4'
```
Comment (Lameck1) [inline Gemfile]: - [ ] In line with the above comment, this **`sqlite3`** gem should now be removed. Instead, we should now be using the **`pg`** gem for the PostgreSQL database. Please consider this.

File: config/routes.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+Rails.application.routes.draw do
+  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  get '/users' , to: 'users#index';
+  get '/users/:id' , to: 'users#show';
+  get '/users/:id/posts' , to: 'posts#index';
+  get '/users/:id/posts/:id' , to: 'posts#show';
```
Comment (Lameck1) [inline config/routes.rb]: - [ ] We can rewrite our routes in a more RESTFUL way to achieve the desired URLs mentioned in the project requirements. When I run `rails routes` the result of the URLs I get do not match the ones provided in the project requirements. See below:

 | **URLs from project requirements** | **URLs from your project** |
 | --- | --- |
 | image | image |
 
 Please consider rewriting the routes as below:
 
 ```ruby
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```

**Review by Lameck1 — CHANGES_REQUESTED** [review]
Hi @DyaryRaoof, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Dipeshtwis — APPROVED** [review]
# Hi @DyaryRaoof, 

### Status: Approved ✔️

:+1:  You have done well implementing the suggested changes. Well Done.

Your project is complete!  There are some optional changes I've suggested to you to make your project best, other than that nothing else to say.
Congratulations! 🎉 


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider: [ Highly Recommended ]
- I would recommend adding a proper PR description so that we would know what features you have added.


**You may close this Pull Request and merge the changes**

  
**Keep being awesome devs and good luck with your next milestone!**
-----

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Dipesh Kumar_**

------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**